### PR TITLE
CI: add workflow to test Makefile

### PR DESCRIPTION
**Pull request by achimcc**
Given recent problems reported in https://github.com/paritytech/ink-playground/issues/727 and https://github.com/paritytech/ink-playground/issues/728, I decided it makes sense to add a CI workflow which makes sure, that the Makefile commands `install`, `generate`, `build` and `backend-run` always work on latest ubuntu.

### CODE REVIEW

File: .github/workflows/makefile.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Install
+        run: make install
+
+      - name: Install required rust toolchain
+        run: rustup toolchain install nightly-2022-05-24
+
+      - name: Add rust-src to toolchain
+        run: rustup component add rust-src --toolchain nightly-2022-05-24-x86_64-unknown-linux-gnu
```
Comment (JoshOrndorff) [inline .github/workflows/makefile.yml:32]: Can / should the make scripts do this for us? If not, it should be mentioned in the readme to manually run these commands.

Comment (achimcc) [reply .github/workflows/makefile.yml:32]: Yes, those two easily.

`wasm-pack` is a bit more complicated, we don't want to download and run the installer again, if wasm-pack is already installed!

I was planning to do a second PR for all three, but I can move those two to the Makefile now!

Comment (JoshOrndorff) [reply .github/workflows/makefile.yml]: Yeah, I agree wasm-pack is more complex, and can be handled later.

Comment (achimcc) [reply .github/workflows/makefile.yml]: I moved those two to the Makefile for now.

File: .github/workflows/makefile.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt
+
+      - uses: actions/checkout@v3
+
+      - name: Install
```
Comment (JoshOrndorff) [inline .github/workflows/makefile.yml:25]:
```suggestion
      - name: Check `make install` command
```

So it is consistent with the others

**Pull request merged**